### PR TITLE
Abris #242 remove dependency on confluent naming strategy code

### LINKED ISSUE #242: Remove dependency on confluent subject naming strategy code
The code for generating subject name was moved between confluent versions, and it is causing issues for us.

Since the code is really simple (just concatenation of Strings, really). I think it's better to implemented yourself in Abris and avoid this problematic dependency.

### PR DESCRIPTION
**Pull request by cerveada**
fixes #242

### CODE REVIEW

File: src/main/scala/za/co/absa/abris/avro/registry/SchemaSubject.scala [hunk 1]
```diff
@@ -30,31 +30,27 @@ class SchemaSubject(val asString: String) {
...
-  private val TOPIC_RECORD_NAME_STRATEGY = new TopicRecordNameStrategy()
-
 
   def usingTopicNameStrategy(
     topicName: String,
     isKey: Boolean = false
   ): SchemaSubject = {
-    val dummySchema = createDummySchema("name", "namespace")
-    new SchemaSubject(TOPIC_NAME_STRATEGY.subjectName(topicName, isKey, dummySchema))
+    val suffix = if (isKey) "-key" else "-value"
```
Comment (kevinwallimann) [inline src/main/scala/za/co/absa/abris/avro/registry/SchemaSubject.scala:38]: Maybe add a comment here, with a reference to the `TopicNameStrategy` implementation from confluent, just so it's transparent where the `-key` and `-value` convention comes from.

File: src/main/scala/za/co/absa/abris/avro/registry/SchemaSubject.scala [hunk 2]
```diff
@@ -63,16 +59,23 @@ object SchemaSubject{
...
   ): SchemaSubject = {
-    new SchemaSubject(TOPIC_RECORD_NAME_STRATEGY.subjectName(topicName, false, schema))
+    new SchemaSubject(topicName + "-" + getRecordName(schema))
   }
 
+  private def getRecordName(schema: Schema): String =
+    if (schema.getType == Type.RECORD) {
+      schema.getFullName
+    } else {
+      throw new IllegalArgumentException("Schema must be of type RECORD")
```
Comment (kevinwallimann) [inline src/main/scala/za/co/absa/abris/avro/registry/SchemaSubject.scala:76]: Please add the actual type to the error message

**Comment by kevinwallimann** [comment]
The naming code should really never be changed. Otherwise, new schemas could be registered even if they already exist. Would it make sense to extract the part that should never be changed to a separate class?

File: src/main/scala/za/co/absa/abris/avro/registry/SchemaSubject.scala (hunk 1 above)
Comment (cerveada) [reply src/main/scala/za/co/absa/abris/avro/registry/SchemaSubject.scala:38]: Is the link on top of SchemaSubject class enough, or should there be a special comment here as well?
The link is hidden in the diff, so I don't know if you saw it.

**Comment by cerveada** [comment]
> The naming code should really never be changed. Otherwise, new schemas could be registered even if they already exist. Would it make sense to extract the part that should never be changed to a separate class?

I don't think there is much else in the class. It's already the class main and only responsibility to create the subject name. So it does not make much sense to me to create another class.

File: src/main/scala/za/co/absa/abris/avro/registry/SchemaSubject.scala (hunk 2 above)
Comment (cerveada) [reply src/main/scala/za/co/absa/abris/avro/registry/SchemaSubject.scala]: done

File: src/main/scala/za/co/absa/abris/avro/registry/SchemaSubject.scala (hunk 1 above)
Comment (kevinwallimann) [reply src/main/scala/za/co/absa/abris/avro/registry/SchemaSubject.scala:38]: Ah ok, then it's fine. No, I didn't see it

**Review by kevinwallimann — APPROVED** [review]

**Pull request merged**